src/python/grpcio/grpc/_auth.py

```diff
@@ -60,15 +69,7 @@ def __call__(self, context, callback):
         else:
             future = self._pool.submit(self._credentials.get_access_token)
         future.add_done_callback(
-            lambda x: self._get_token_callback(callback, x))
-
-    def _get_token_callback(self, callback, future):
-        try:
-            access_token = future.result().access_token
-        except Exception as e:
-            _sign_request(callback, None, e)
-        else:
-            _sign_request(callback, access_token, None)
+            lambda ignored_future: _get_token_callback(callback, future))
```

Comment:
The original version

    lambda x: self._get_token_callback(callback, x)

without the `self` is what I have in mind. Is there any problem with that that I can't see? I'd argue even if the current structure passes the future around superfluously, capturing one additional variable makes the code less robust relative to using the argument it passes you, expanding the effective scope of the local variable `future` [and it's a tiny bit less efficient too—one needs to carry a reference to the variable closed over.]